test/core/modules/benchmark.js

```diff
@@ -0,0 +1,31 @@
+
+var imageName = 'image1';
+sequencer.loadImages(imageName, image);
+
+test('benchmark all modules',function(t){
+
+  var start, end;
+
+  console.log("############ Benchmarks ############");
+
+  Object.keys(sequencer.modules).forEach(function forEachModule(moduleName) {
+    start = Date.now();
+    sequencer.addSteps(moduleName);
```

Comment:
I see - we aren't actually running the sequencer. And even when we do, we need to do something like this to measure the time:

```js
sequencer.run(function runSequencer() {
  end = Date.now();
  console.log("Module " + moduleName + " ran in: " + (end-start) + " milliseconds");
});
```

Should we look at these asynchronous iterator patterns?

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for-await...of

https://stackoverflow.com/questions/4288759/asynchronous-for-cycle-in-javascript#4288992

???

Reply:
@publiclab/is-reviewers what do you think? Anyone have some experience with asynchronous iterators, or have another simpler way to do this? I think we have to ensure that the modules don't run at the same time or our benchmarks won't really be fair. Each step should run and be timed by itself.

Reply:
If anyone wants to submit a new PR based on this, that'd be great!

Reply:
@jywarren can I give it a shot ?